### ISSUE TITLE
Problem with @Indexed annotation

### ISSUE THREAD

**Issue by wromijn** [flapdoodle-oss/de.flapdoodle.embed.mongo#195]
I found a weird issue when using embedded Mongo in a Spring Boot project. I want to use a real Mongo server in production and embedded Mongo for testing. This works fine: running the tests throws an exception about not being able to connect to the real Mongo server, but they run without problems.

However, as soon as I add an @Indexed annotation to one of the fields of the entity that is persisted in Mongo, it seems that embedded Mongo is not used anymore. Spring tries to connect to the real Mongo server and times out.

I created a minimal project to illustrate the issue:
https://github.com/wromijn/embedded-mongo-problem

The test breaks, but passes when the @Indexed annotation on the name field of the Person entity is removed. Interestingly, removing spring-boot-starter-web from the pom.xml file also makes the test pass.

**Comment by Loki-Afro** [comment]
Well ... there is no issue or does SpringBootTest say in any way that it will start a mongod instance? I don't think so :)

Take a look here http://stackoverflow.com/a/31568654/1248724

You don't have to change your dependencies, they are allright, you just have to provide This Test Config and Add it to your test Application Context.

**Comment by wromijn** [comment]
Oh wow! I never thought it would be this easy! You're right, the Spring documentation isn't very clear on this. Thank you very much for your help!

**Comment by acguglielmo** [comment]
I'm facing the same problem... what was the solution? The PR link in @Loki-Afro comment is broken.

**Comment by wromijn** [comment]
> I'm facing the same problem... what was the solution? The PR link in @Loki-Afro comment is broken.

The problem was that I created a MongoDbFactory bean in a configuration class and always applied this configuration, regardless whether the application or the tests were run. So when running the tests, the embedded MongoDb wasn't even loaded, because the normal MongoDb was already there. Removing the configuration class (which didn't add anything to the default configuration) solved the problem: https://github.com/wromijn/embedded-mongo-problem/compare/solution

The real question is: why didn't my misguided approach lead to problems before the @Indexed annotation was added? :)
